### PR TITLE
Better error message for invalid parameters in sorter

### PR DESCRIPTION
**Pull request by h-mayorquin**
When the error is raised it tells the user what went wrong and what was expected. 

Related to #3155

### CODE REVIEW

File: src/spikeinterface/sorters/basesorter.py [hunk 1]
```diff
@@ -169,14 +169,12 @@ def params_description(cls):
     @classmethod
     def set_params_to_folder(cls, recording, output_folder, new_params, verbose):
         params = cls.default_params()
+        valid_parameters = cls.params_description().keys()
```
Comment (zm711) [inline src/spikeinterface/sorters/basesorter.py:172]: You've already got the params above. Why not just make valid params from

params.keys()

Wouldn't that be clearer than pulling in the descriptions dict and using those keys?

Comment (h-mayorquin) [reply src/spikeinterface/sorters/basesorter.py:172]: Yes. Agree. Changing.

Comment (h-mayorquin) [reply src/spikeinterface/sorters/basesorter.py:172]: Done.

**Review by samuelgarcia — APPROVED** [review]

**Comment by samuelgarcia** [comment]
merci.

**Pull request merged**